src-electron/upgrade/upgrade.js

```diff
@@ -40,14 +48,48 @@ async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
   zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data
+  if (isJsonString(obj)) {
+    data = JSON.parse(obj)
+  }
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * Returns a list of attributes with proper global attributes
+ *
+ * @export
+ * @param {*} db
+ * @param {*} clusterRef
+ * @param {*} attributes
+ * @returns attributes
+ */
+
+async function computeStorageTemplate(db, clusterRef, attributes) {
+  let clusterName
+  let forcedExternal
+  clusterName = await queryCluster.selectClusterName(db, clusterRef)
+  return await Promise.all(
+    await attributes.map(async (attribute) => {
+      forcedExternal = await getForcedExternalStorage(db, attribute.id)
+      if (
+        forcedExternal.byName &&
+        forcedExternal.byName[clusterName] &&
+        forcedExternal.byName[clusterName].includes(attribute.name)
+      ) {
+        attribute.storagePolicy = dbEnum.storagePolicy.attributeAccessInterface
+        return attribute
+      } else {
+        return attribute
```

Comment:
Why not just `return attribute` outside the `if`?

Reply:
agree